cloudflare/resource_cloudflare_api_token.go

```diff
@@ -0,0 +1,248 @@
+package cloudflare
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceCloudflareApiToken() *schema.Resource {
+	p := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"resources": {
+				Type:     schema.TypeMap,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"permission_groups": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"effect": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "allow",
+				ValidateFunc: validation.StringInSlice([]string{"allow", "deny"}, false),
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceCloudflareApiTokenCreate,
+		Read:   resourceCloudflareApiTokenRead,
+		Update: resourceCloudflareApiTokenUpdate,
+		Delete: resourceCloudflareApiTokenDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &p,
+			},
+			"request_ip_in": {
```

Comment:
While right now, IP is the only policy condition, we should update the schema to match the API response structure to allow for more conditions in the future should another one be added. Something like

```hcl
condition {
  ip {
    not_in = ["192.0.2.1"]
    in = ["198.51.100.0"]
  }
}
```